### PR TITLE
fixed typo

### PR DESCRIPTION
**Pull request by AbhineshJha**
Some words are corrected in right manner and also some grammar mistakes .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -216,7 +216,7 @@ com.example.myapplication.HasNamedCompanion2
 
 
 <details>
-<summary>Example of named companion rules for R8 full mode</summary>
+<summary>Example of named companion rules for R8 full-mode</summary>
```
Comment (sandwwraith) [inline README.md:219]: 'Full mode' is mentioned in Google documentation without a dash:
https://developer.android.com/build/shrink-code#full-mode
https://r8.googlesource.com/r8/+/refs/heads/master/compatibility-faq.md#r8-full-mode

I prefer being consistent with them.

File: README.md [hunk 2]
```diff
@@ -180,11 +180,11 @@ dependencies {
 }
 ```
 
->We also provide `kotlinx-serialization-core` artifact that contains all serialization API but does not have bundled serialization format with it
+>We also provide `kotlinx-serialization-core` artifacts that contains all serialization API but does not have bundled serialization format with it
```
Comment (sandwwraith) [inline README.md:183]: While technically, several core artifacts exist (for jvm, native, etc...), ideologically, they are all the same. Moreover, platform artifacts have different coordinates (`kotlinx-serialization-core-jvm`, etc). From a user standpoint, only one `kotlinx-serialization-core` should be used. Therefore, I'd prefer to leave this word in a singular form.

Also, you've had to change `contains` to match `artifacts`.

**Review by sandwwraith — CHANGES_REQUESTED** [review]

**Comment by AbhineshJha** [comment]
ok as your wish !

File: README.md (hunk 2 above)
Comment (AbhineshJha) [reply README.md:183]: ok as your wish

**Pull request closed without merging**